apps/web-connect/src/features/checkout/components/ActionSection.tsx

```diff
@@ -50,7 +50,7 @@ export function ActionSection(): JSX.Element {
      */
     const getTokenButtonText = useCallback(() => {
         if (mintWithEth.isPending || mintWithXai.isPending || approve.isPending) return "WAITING FOR CONFIRMATION..";
-        if (!isValidNetwork(chain?.id, isDevelopment)) return "Please Switch to Arbitrum";
+        if (!isValidNetwork(chain?.id as number, isDevelopment)) return "Please Switch to Arbitrum";
```

Comment:
It seems to me we should handle the new type in `isValidNetwork` rather than adding the `as number` to all the usages.

Reply:
I agree, see my other comment for the updates.